src/openswathalgo/include/OpenMS/OPENSWATHALGO/ALGO/Scoring.h

```diff
@@ -131,10 +131,14 @@ namespace OpenSwath
     OPENSWATHALGO_DLLAPI void normalize_sum(double x[], unsigned int n);
 
     // Compute rank of vector elements
-    OPENSWATHALGO_DLLAPI std::vector<unsigned int> computeRank(const std::vector<double>& w);
+    OPENSWATHALGO_DLLAPI void computeRank(const std::vector<double>& v, std::vector<unsigned int>& ranks);
+
+    // Compute rank of vector elements for each row in a 2D array
+    OPENSWATHALGO_DLLAPI void computeRankVector(const std::vector<std::vector<double>>& intensity, std::vector<std::vector<unsigned int>>& ranks);
 
     // Estimate rank-transformed mutual information between two vectors of data points
-    OPENSWATHALGO_DLLAPI double rankedMutualInformation(std::vector<double>& data1, std::vector<double>& data2);
+    //OPENSWATHALGO_DLLAPI double rankedMutualInformation(std::vector<double>& data1, std::vector<double>& data2);
+    OPENSWATHALGO_DLLAPI double rankedMutualInformation(std::vector<unsigned int>& data1, std::vector<unsigned int>& data2);
```

Comment:
add some documentation that the input vectors must be ranks.
Name the parameters `ranked_data1` and `ranked_data2`